### PR TITLE
monero-gui: only exclude darwin in meta.platforms

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
The platforms have been restricted to x86_64-linux in the past because of failures for darwin and aarch64. However, I have just tried, aarch64 as well as i686 build now.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change (monero-gui)
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jonringer** [comment]
unfortunately qtwebengine is broken on darwin, so it's still accurate to exclude it :(

File: pkgs/applications/altcoins/monero-gui/default.nix [hunk 1]
```diff
@@ -105,7 +105,7 @@ stdenv.mkDerivation rec {
     description = "Private, secure, untraceable currency";
     homepage    = https://getmonero.org/;
     license     = licenses.bsd3;
-    platforms   = [ "x86_64-linux" ];
+    platforms   = filter (p: p != "x86_64-darwin") platforms.all;
```
Comment (LnL7) [inline pkgs/applications/altcoins/monero-gui/default.nix]: you can use meta.badPlatforms to do the same declaratively.

Comment (rnhmjoj) [reply pkgs/applications/altcoins/monero-gui/default.nix]: Thank you, I didn't know about it.

**Comment by rnhmjoj** [comment]
@jonringer ah, that's why. Is there an issue tracking this?

**Comment by jonringer** [comment]
@rnhmjoj I have a 2012 Macbook Pro laying around and i tried doing the build, that's the package that it broke on. I'm personally not commited enough to darwin to try and fix it myself.

**Comment by rnhmjoj** [comment]
I have just found out some of the patches weren't needed anymore (translations are finally embedded in the binary).

**Comment by rnhmjoj** [comment]
ping

**Comment by rnhmjoj** [comment]
Could this be merged?

**Comment by mmahut** [comment]
@GrahamcOfBorg build monero-gui

**Pull request merged**